Allwmake

```diff
@@ -97,8 +98,11 @@ else
     if [ "${ADAPTER_WMAKE_UNDEFINED_SYMBOLS}" -gt 0 ] || [ "${ADAPTER_LD_UNDEFINED_SYMBOLS}" -gt 0 ]; then
         log "=== ERROR: Building completed with linking problems: there were undefined symbols. ==="
         log "Possible causes:"
-        log "- Is preCICE discoverable at runtime? Check the content of ADAPTER_PKG_CONFIG_LIBS above."
+        log "- Is preCICE discoverable at runtime? Check the content of pkg-config output above."
```

Comment:
I think users recently also faced the issue of pkg-config not being installed. Since the script relies on pkg-config, we may want to add a safety check for this as well. Also, what is the pkg-config output supposed to be? How can I fix it in case it doesn't show the right thing?

Reply:
Good point about not having pkg-config installed. I will try to catch this (rare) case in the same PR.

This is supposed to be the pkg-config output, should I refer to it in some other way?

```
log "If not already known by the system, preCICE may be located using:"
log "  pkg-config --cflags libprecice  = ${ADAPTER_PKG_CONFIG_CFLAGS}"
log "  pkg-config --libs libprecice    = ${ADAPTER_PKG_CONFIG_LIBS}"
```

Reply:
no, this is set automatically and is not in the head of the script. Here I just copied the log part of the script, normally you would get paths and options there (at least a `-lprecice`).

Reply:
I added a check if pkg-config is installed and I throw a warning, if not. In principle, `pkg-config` is not an always required dependency, as there are also other ways to make preCICE discoverable.

I checked this by moving `/usr/bin/pkg-config`. This is the log I get:

```
Building the OpenFOAM-preCICE adapter...
WARNING: pkg-config configuration file 'libprecice.pc' was not found in the PKG_CONFIG_PATH.
        Check the preCICE documentation page 'Linking to preCICE' for more details, or proceed if you know what you are doing.

Current OpenFOAM environment:
  WM_PROJECT = OpenFOAM
  WM_PROJECT_VERSION = v2112

The adapter will be built into /home/makish/OpenFOAM/makish-v2112/platforms/linux64GccDPInt32Opt/lib
Additional preprocessor/compiler options: 

Building with WMake (see the wmake.log log file)...

Compiling enabled on 4 cores
wmake libso (openfoam-adapter)
Everything looks fine in wmake.log.
Everything looks fine in ldd.log.
=== OK: Building completed successfully! ===
There have been warnings. Check the log above.
```

I also implemented a new function for warnings, just to reduce duplication. I could also do this for errors, but that's already getting into a different direction.

Reply:
@davidscn I have addressed all your comments in this conversation:
- This works both for users with and without pkg-config, as well as with discoverable and non-discoverable `libprecice.pc`.
- If any of these is not found, the user gets a warning that essentially says "it's fine if building works".

Please check and approve, if ok.

Reply:
Looks good now.